### DISCUSSION TITLE
Manual leveling Inconsistency - Z height change

### DISCUSSION THREAD

**Discussion by NpFaith** [Jyers/Marlin#1274] [Q&A] [answered]
Hi, I need help trying to figure out what's going on with my Ender 3 v2, the method I use to level the bed is manual levelling what's in the prepare tab, what happens after homing and starting the process, I notice that whenever I click to change position he keeps tightening more without me moving the pulleys, I do homing and he gets loose again. 

It seems to me that whenever I choose the position the Z axis decreases. If I home before moving to the next position it maintains the Z height.

Is there a a way to add gcode on the firmware before when I click the bottom left, top left, etc it homing automatic? Thanks for your time.

**Comment by LightEmittingDodo** [comment, marked as answer]
Having the same issue. I've been looking for solutions for a while now. If you keep pressing the same button for a manual leveling position it takes the z deeper and deeper each time. 
There are also the same issues in manual mesh leveling where the z height is inconsistent even on the same point without touch the leveling. 
I've had to go back on stock for now. 
I tried another custom fw (mirsoc I think) which has the exact same issue.
How to replicate: Go to manual leveling. Press any of the positions. Check with paper but don't touch the leveling. Press the button for the same position so the nozzle rises and falls back. Z will be further down now than it was before. Repeat and crack bed.

**Reply by NpFaith** [reply]
For a moment I thought it was just my printer's problem but apparently I'm not the only one.

I am a mere user of this firmware and I am very grateful to those who developed it however there are flaws but I don't know if the problem is not the hardware itself. What I do to get a perfect alignment is:

-Home all
-Manual levelling - Bottom left
-Home all
-Manual levelling - Top left
-Home all
-Manual levelling - Top right
-Home all
-Manual levelling - Bottom right
-Do 2 to 3 times
-Then when the printing starts, go to Z-Offset and drop until it reaches the perfect distance.

I know it takes a lot of work but it was the only way I could make it work.

**Reply by NpFaith** [reply]
I use this one E3V2-ManualMesh-3x3-v4.2.2-v1.3.5.bin. Thanks

**Reply by LightEmittingDodo** [reply]
> @NpFaith @LightEmittingDodo
> Can you tell us your firmware versions?

I have this problem  on 1.3.5b, 1.3.5 and 1.3.4. 3x3 mesh and default for all. I didn't go back and try previous ones.
A bit more info: Stock ender 3 V2 . 4.4.2. No probe.
Leveling manually by homing,  disabling steppers and then paper method results in a nozzle that's too high during print. Same results with mesh. I can print fine on stock FWs with manual leveling though.

**Reply by NpFaith** [reply]
LightEmittingDodo I never try it with stock nice to know that works fine. Thanks for the info.

**Reply by LightEmittingDodo** [reply]
> @NpFaith @LightEmittingDodo
> It seems to me that it is rather a problem of methodology.
> The Manual Mesh firmware involves the same methodology as its equivalent with BLTouch ... including at the StartGcode level.
> 
> In the stock version, you don't have a Z-Offset like the one in the Manual Mesh version.
> The homing will be different ...

Just for clarity, by stock I mean what's on the Creality website. 1.0.2.
I think the primary problem here is using the manual leveling (not mesh) the z axis moves closer and closer to the bed without any manual adjustment to the bed, which makes it impossible to manually level it. This is when keeping z offset to 0.
Perhaps me and NpFaith are making the same mistakes in leveling the bed. I'm not able to pinpoint the exact problem. 
How is the homing different in the stock version?

**Reply by LightEmittingDodo** [reply]
So I tried manual mesh again. The issue remains of going to the same point multiple times without changing the z offset in manual tuning results in the nozzle getting closer and closer to the bed each time. It should be coming back to the same position if I'm not wrong. The z pos shown on screen remains the same. Very odd.

**Reply by NpFaith** [reply]
I tried this one E3V2-Default-v4.2.2-v1.3.5.b.bin and its the same... :(

**Reply by NpFaith** [reply]
I have my ender 3 v2 with practically stock hardware, it only has the new springs, the yellow ones that everyone uses. What I use to test is thin paper and I adjust the bed knob until it rubs a little, after I press the option Bottom left and I move the paper, after repeating a few times, the paper folds and gets tighter under the nozzle without touching the bed. Perhaps there is a difference in the movement of the motor or moving parts that makes the board think it is in the same position and it is not. This problem makes it very difficult to balance the bed well. I need to test with the official firmware but I really like Jyers, but is life you cant have it all.

**Reply by LightEmittingDodo** [reply]
What I'm current using as a workaround is to do the following:
1. Manually level bed by homing, disabling steppers and paper method.
2. Getting rough mesh with paper method in manual mesh mode
3. Printing a leveling test, and live adjusting z offset. Taking mental notes here on which side to adjust and how in mesh level
4. Re adjust mesh according to results of the previous test print
5. Test print again. Adjusting z offset if needed. 
This has gotten me close to something acceptable until I can change my warped bed and get a BLtouch

**Reply by NpFaith** [reply]
BLtouch solves allot problems. More things to buy... new board 4.2.7 + Bltouch + Allmetal hotend + metal extruder, if I knew I had to buy all this, I should have bought ender 3 V1 instead of V2 and spent the difference on upgrades. Walking and learning!

**Reply by LightEmittingDodo** [reply]
Thing is manual mesh is supposed to be good enough to not have to buy BLtouch other than for convenience. But I think my problem might lie in my warped bed because I'm having to adjust z offset at every print. And that is why I wanted manual mesh in the first place lol.
As with any hobby, there's always an upgrade weighing on your wallet. 🥲

**Reply by NpFaith** [reply]
tititopher68-dev you are right Endstop switch will never have precision. Thanks for the help

**Reply by LightEmittingDodo** [reply]
@tititopher68-dev you're very right regarding temps. Also the fact that I have a warped bed comes into play here which results in it expanding slightly differently each time I heat it up. The mechanical Z endstop is highly inaccurate especially with how it's installed in stock enders.
Honestly so much can be at fault here I'm just looking for whatever's going to work for me. But nonetheless, with the manual leveling I'm almost sure it's a marlin bug with how consistently it behaves lol. Even in other custom FWs.
Edit: I think I should mention I do all leveling, z offset etc at hot temps. 65 bed, 170 nozzle.

**Reply by NpFaith** [reply]
@tititopher68-dev I was searching and I find Endstop phase there exist in other firmwares, it is possible to add on to the future firmware? Thanks for your time

**Reply by NpFaith** [reply]
@tititopher68-dev 
Allot of variables... thanks for everything explain here and for the help
